jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/bigquery/BigQueryQueryRewriter.java

```diff
@@ -0,0 +1,161 @@
+package com.exasol.adapter.dialects.bigquery;
+
+import java.sql.*;
+import java.util.logging.*;
+import java.util.regex.*;
+
+import com.exasol.*;
+import com.exasol.adapter.*;
+import com.exasol.adapter.dialects.*;
+import com.exasol.adapter.jdbc.*;
+import com.exasol.adapter.sql.*;
+
+/**
+ * This class implements a BigQuery-specific query rewriter
+ */
+public class BigQueryQueryRewriter extends BaseQueryRewriter {
+    private static final Logger LOGGER = Logger.getLogger(BigQueryQueryRewriter.class.getName());
+    private static final double[] TEN_POWERS = { 10d, 100d, 1000d, 10000d, 100000d, 1000000d };
+    private static final Pattern DATE_PATTERN = Pattern.compile("(\\d{4})-(\\d{1,2})-(\\d{1,2})");
+    private static final Pattern TIME_PATTERN = Pattern.compile("(\\d{1,2}):(\\d{1,2}):(\\d{1,2})(?:\\.(\\d{1,6})?)?");
```

Comment:
```suggestion
    private static final Pattern TIME_PATTERN = Pattern.compile("(\\d{1,2}):(\\d{1,2}):(\\d{1,2})(?:\\.(\\d{1,6}))?");
```
The milliseconds are there either including the dot or not at all.

Reply:
fixed